gobblin-utility/src/main/java/gobblin/util/PathUtils.java

```diff
@@ -177,4 +177,14 @@ public static void deleteEmptyParentDirectories(FileSystem fs, Path limitPath, P
       deleteEmptyParentDirectories(fs, limitPath, startPath.getParent());
     }
   }
+
+  /**
+   * Add a prefix into a path, like "file:" when comparing two Paths,
+   * to avoid the case when the actual path are the same while their comparison result turns out to be false.
+   * @param path
+   * @param prefix
+   */
+  public static Path addFilePrefixForComparison( Path path, String prefix ) {
```

Comment:
I think the correct way is to strip both files of scheme and authority instead, e.g. `PathUtils.getPathWithoutSchemeAndAuthority(path1).equals(PahtUtils.getPathWithoutSchemeAndAuthority(path2))`